File: webui/core/src/main/java/org/exoplatform/webui/core/UIToolbar.java
/**
 * Copyright (C) 2009 eXo Platform SAS.
 * 
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation; either version 2.1 of
 * the License, or (at your option) any later version.
 * 
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */

package org.exoplatform.webui.core;

import java.util.List;

/**
 * Created by The eXo Platform SARL
 * Author : Dang Van Minh
 *          minhdv@exoplatform.com
 * Aug 31, 2006  
 * 
 * A component that represents a toolbar
 */

public abstract class UIToolbar extends UIComponent
{
   /**
    * The css style
    */
   private String toolbarStyle_ = "LightToolbar";

   /**
    * A javascript expression
    */
   private String strJavascript_ = "";

   /**
    * A javascript event name
    */
   private String eventName_ = "";

   public UIToolbar() throws Exception
   {
   }

   public String getToolbarStyle()
   {
      return toolbarStyle_;
   }

   public void setToolbarStyle(String toolbarStyle)
   {
      toolbarStyle_ = toolbarStyle;
   }

   public List getEvents()
   {
      return getComponentConfig().getEvents();
   }

   public String getJavascript()
   {
      return strJavascript_;
   }

   public String getEventName()
   {
      return eventName_;
   }

   public void setJavascript(String eventName, String strJavascript)
   {
      strJavascript_ = strJavascript;
      eventName_ = eventName;
   }

}

File: web/eXoResources/src/main/webapp/javascript/eXo/webui/UIToolbar.js
/**
 * Copyright (C) 2009 eXo Platform SAS.
 * 
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation; either version 2.1 of
 * the License, or (at your option) any later version.
 * 
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */

function UIToolbar() {};
/**
 * Display clicked element on block state
 * @param {Object} clickedEle clicked object
 */
UIToolbar.prototype.displayBlockContent = function(clickedEle) {
	if(clickedEle == null) return;
  var uiToolbar = eXo.core.DOMUtil.findAncestorByClass(clickedEle, "UIToolbar");
  var parentUIToolbar = uiToolbar.parentNode;
  
  var buttons = eXo.core.DOMUtil.findDescendantsByClass(uiToolbar, "div", "Button");
  var buttonLabel = eXo.core.DOMUtil.findDescendantsByClass(uiToolbar, "div", "ButtonLabel");

  var uiToolbarContentContainer = eXo.core.DOMUtil.findFirstDescendantByClass(parentUIToolbar, "div", "UIToolbarContentContainer");
  var uiToolbarContentBlock = eXo.core.DOMUtil.getChildrenByTagName(uiToolbarContentContainer, "div");

  for(var i = 0; i < buttons.length; i++) {
    if(clickedEle == buttons[i]) {
      uiToolbarContentBlock[i].style.display = "block";
      buttonLabel[i].style.fontWeight = "bold";
    } else {
      uiToolbarContentBlock[i].style.display = "none";
      buttonLabel[i].style.fontWeight = "100";		
    }                                                                  
  }
  
};

eXo.webui.UIToolbar = new UIToolbar();
